sdk/typespec/typespec_client_core/src/http/mod.rs

```diff
@@ -22,12 +24,23 @@ pub use options::*;
 pub use pageable::*;
 pub use pipeline::*;
 pub use request::{Body, Request, RequestContent};
-pub use response::{Model, Response};
+pub use response::Response;
+pub use response_body::ResponseBody;
+pub use response_future::{LazyResponse, ResponseFuture};
 
 // Re-export important types.
 pub use http_types::{Method, StatusCode};
 pub use url::Url;
 
+use bytes::Bytes;
+use futures::Stream;
+use std::pin::Pin;
+
+#[cfg(not(target_arch = "wasm32"))]
+pub type PinnedStream = Pin<Box<dyn Stream<Item = crate::Result<Bytes>> + Send + Sync>>;
+#[cfg(target_arch = "wasm32")]
+pub type PinnedStream = Pin<Box<dyn Stream<Item = crate::Result<Bytes>>>>;
```

Comment:
Moved these types here because they are useful across a few different modules in `http`. This does mean it's been promoted from `typespec_client_core::http::response::PinnedStream` to `typespec_client_core::http::PinnedStream`